caches/infinispan/src/main/java/org/commonjava/maven/galley/cache/infinispan/FastLocalCacheProvider.java

```diff
@@ -100,6 +101,12 @@
 
     private final JoinableFileManager fileManager = new JoinableFileManager();
 
+    // This file counter is used to solve the problem of multi file operations in on ISPN TX. Sometimes in one single ISPN TX,
+    // it may includes more than one file operations. If we don't control the ISPN lock of each operation separately and just use
+    // tx.commit/rollback, it may cause ISPN TX in some weird state. This file counter is more like a thread re-entrant feature for
+    // ISPN single TX for different files.
+    private final AtomicInteger fileCounter = new AtomicInteger( 0 );
```

Comment:
Won't this cross-pollute between paths, such that two threads accessing different paths will use the same counter, and only one of the paths' transactions will be closed?

* Thread Context 1: open file (increment counter == 1)
* Thread Context 2: open file2 (increment same counter == 2)
* Thread Context 2: close file2 (decrement counter == 1; file2 transaction NOT closed)
* Thread Context 1: close file( decrement counter == 0; close file transaction)

Should this counter be stored in ThreadContext?